docs/src/rules/no-restricted-imports.md

```diff
@@ -20,176 +20,139 @@ It applies to static imports only, not dynamic ones.
 
 ```json
 "no-restricted-imports": ["error", {
-    "patterns": [{
-      "group": ["import-foo/*"],
-      "importNamePattern": "^foo",
-    }]
+    "name": "import-foo",
+    "message": "Please use import-bar instead."
+}, {
+    "name": "import-baz",
+    "message": "Please use import-quux instead."
 }]
 ```
 
-To restrict the use of all Node.js core imports (via <https://github.com/nodejs/node/tree/master/lib>):
-
-```json
-    "no-restricted-imports": ["error",
-         "assert","buffer","child_process","cluster","crypto","dgram","dns","domain","events","freelist","fs","http","https","module","net","os","path","punycode","querystring","readline","repl","smalloc","stream","string_decoder","sys","timers","tls","tracing","tty","url","util","vm","zlib"
-    ],
-```
-
-## Examples
-
-Examples of **incorrect** code for this rule:
+Examples of **incorrect** code for string option:
 
 ::: incorrect { "sourceType": "module" }
 
 ```js
-/*eslint no-restricted-imports: ["error", "fs"]*/
+/*eslint no-restricted-imports: ["error", {
+    "name": "disallowed-import",
+    "message": "Please use 'allowed-import' instead"
+}]*/
 
-import fs from 'fs';
+import foo from 'disallowed-import';
 ```
 
 :::
 
-::: incorrect { "sourceType": "module" }
+### paths
 
-```js
-/*eslint no-restricted-imports: ["error", "fs"]*/
+This is an object option whose value is an array and the name of the module you want to restrict can be specified inside this array, same as in string option above.
 
-export { fs } from 'fs';
+```json
+"no-restricted-imports": ["error", { "paths": ["import1", "import2"] }]
 ```
 
-:::
+Examples of **incorrect** code for `paths`:
 
 ::: incorrect { "sourceType": "module" }
 
 ```js
-/*eslint no-restricted-imports: ["error", "fs"]*/
+/*eslint no-restricted-imports: ["error", { "paths": ["cluster"] }]*/
 
-export * from 'fs';
+import cluster from 'cluster';
 ```
 
 :::
 
-::: incorrect { "sourceType": "module" }
+Custom messages for a particular module can also be specified in `paths` array option using objects as above.
 
-```js
-/*eslint no-restricted-imports: ["error", { "paths": ["cluster"] }]*/
-
-import cluster from 'cluster';
+```json
+"no-restricted-imports": ["error", {
+    "paths": [{
+        "name": "import-foo",
+        "message": "Please use import-bar instead."
+    }, {
+        "name": "import-baz",
+        "message": "Please use import-quux instead."
+    }]
+}]
 ```
 
-:::
+#### importNames
 
-::: incorrect { "sourceType": "module" }
+This option in `paths` is an array and can be used to specify certain imports to restrict that are being exported from a module. Import names specified inside `paths` array looks for the module specified in the `name` property of corresponding object, so it is required to specify the `name` key with both `importNames` and `message` options.
```

Comment:
```suggestion
This option in `paths` is an array and can be used to specify the names of certain bindings exported from a module. Import names specified inside `paths` array affect the module specified in the `name` property of corresponding object, so it is required to specify the `name` key with both `importNames` and `message` options.
```

Question: is `message` actually required when using `importNames`? It doesn't seem like it should be so just double-checking.

Reply:
> it is required to specify the `name` key with both `importNames` and `message` options.

For me this line is saying when using `importNames` or `message` property, then it is require to use `name` property first. I think i should clear it's language a bit.